File: js/src/launcher/NoopBrowserLauncher.ts
import { BrowserLauncher, type LauncherOptions, type LaunchedBrowser } from "./BrowserLauncher.js";

export class NoopBrowserLauncher extends BrowserLauncher {
  async launch(_options: LauncherOptions = {}): Promise<LaunchedBrowser> {
    this.launched = { cdp_url: null, close: async () => {} };
    return this.launched;
  }
}

File: python/modcdp/launcher/NoopBrowserLauncher.py
from __future__ import annotations

from ..launcher.BrowserLauncher import LauncherOptions, BrowserLauncher, LaunchedBrowser


class NoopBrowserLauncher(BrowserLauncher):
    def launch(self, options: LauncherOptions | None = None) -> LaunchedBrowser:
        self.launched = {"cdp_url": None, "close": lambda: None}
        return self.launched
